### PR TITLE
RUMM-3209 Supported versions documentation

### PR DESCRIPTION
**Pull request by maciejburda**
### What and why?

This PR aims to update all the versions and describe the full picture in a new markdown file `SUPPORTED-VERSIONS.md` available in the root of the repo.

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [x] Run unit tests
- [x] Run integration tests
- [x] Run smoke tests

### CODE REVIEW

**Comment by maciejburda** [comment]
I feel like `SUPPORTED-VERSIONS.md` needs another pass. Feel free to suggest things and I'll try to improve it early next week.

File: SUPPORTED-VERSIONS.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+## Platforms
+
+| Platform   | Supported |  Info  |
```
Comment (maxep) [inline SUPPORTED-VERSIONS.md:3]:
```suggestion
| Platform   | Supported |  Version  |
```

File: SUPPORTED-VERSIONS.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+## Networking Compatibility
+| Framework       |   Automatic  | Manual |
+|-----------------|:------------:|:------:|
+| **URLSession**  |       ✅     |   ✅    |
+|[**Alamofire 5+**](https://github.com/DataDog/dd-sdk-ios/tree/develop/Sources/DatadogExtensions/Alamofire) |       ❌     |   ✅    |
+|  **SwiftNIO**   |       ❌     |   ❌    |
+
+## Catalyst
+We support Catalyst in build mode only.
```
Comment (maxep) [inline SUPPORTED-VERSIONS.md:45]: What does 'build mode only' mean in this case?
I think `Mac Catalyst` should be part of the **Platforms** section, no?

File: SUPPORTED-VERSIONS.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+## Dependencies
+We aim to provide dependency-less SDK. Currently there is only one exception to the rule which is PLCrashReporter dependency, required by crash reporting feature.
```
Comment (maxep) [inline SUPPORTED-VERSIONS.md:48]: This section could just show a list. We would just change the phrasing for v2.
```suggestion
The Datatog SDK depends on the following third-party library:
- [PLCrashReporter](https://github.com/microsoft/plcrashreporter) 1.11.0
```

**Review by maxep — COMMENTED** [review]
Thanks, @maciejburda, for gathering all of that 👍 
I've left some suggestions.

File: SUPPORTED-VERSIONS.md (hunk 2 above)
Comment (maciejburda) [reply SUPPORTED-VERSIONS.md:45]: I wanted to keep it separate, because it requires special comment. I'll add a description what build only means :)

**Comment by ncreated** [comment]
I wonder if this should be swift-tools `5.7.1` or `5.7`? From [Apple](https://developer.apple.com/news/?id=jd9wcyov):
> ... Xcode 14.1 or later

and from [Xcode 14.1](https://developer.apple.com/documentation/xcode-release-notes/xcode-14_1-release-notes):
> includes Swift 5.7

**Comment by maciejburda** [comment]
@ncreated I think we should increase the nightly Xcode version to 14.1

**Comment by ncreated** [comment]
> @ncreated I think we should increase the nightly Xcode version to 14.1

As we discussed, we need to first ensure nightly tests won't be broken after this merge. We diagnosed an issue with running them on macOS Ventura (which will be the only host available on Bitrise that can use Xcode 14.1).

**Review by xgouchet — DISMISSED** [review]

**Review by xgouchet — DISMISSED** [review]

**Review by xgouchet — APPROVED** [review]

**Pull request merged**